addons/JobChange/jobchange.lua

```diff
@@ -0,0 +1,145 @@
+_addon.version = '1.0.3'
+_addon.command = 'jc'
+
+-- 1.0.1 first release
+-- 1.0.2 added 'reset' command to simply reset to existing job.  Changes sub job to a random starting job and back.
+-- 1.0.3 Code clean-up
+
+require('tables')
+packets = require('packets')
+res = require ('resources')
+
+function jobchange(job,main_sub)
+    windower.add_to_chat(4,"JobChange: Changing "..main_sub.." job to:"..res.jobs[job].ens)
+    if job and main_sub then 
+        local packet = packets.new('outgoing', 0x100, {
+            [main_sub:sub(1,1):upper()..main_sub:sub(2)..' Job'] = job,
+        })
+        packets.inject(packet)
+        coroutine.sleep(0.5)
+    end    
+end
+
+windower.register_event('addon command', function(command, ...)
+    local self = windower.ffxi.get_player()
+    local args = L{...}
+    local job = ''
+    if args[1] then 
+        job = args[1]:lower()
+    end
+    local main_sub = ''
+    if command:lower() == 'main' then
+        main_sub = 'main'
+    elseif command:lower() == 'sub' then
+        main_sub = 'sub'
+    elseif command:lower() == 'reset' then
+        windower.add_to_chat(4,"JobChange: Resetting Job")
+        main_sub = 'sub'
+        job = windower.ffxi.get_player().sub_job:lower()
+    else
+        windower.add_to_chat(4,"JobChange Syntax: //jc main|sub JOB  -- Chnages main or sub to target JOB")
+        windower.add_to_chat(4,"JobChange Syntax: //jc reset -- Resets Current Job")
+        return
+    end
+    local conflict = find_conflict(job,self)
+    local jobid = find_job(job,self)
+    if jobid then 
+        local npc = find_job_change_npc()
+        if npc then
+            if not conflict then 
+                jobchange(jobid,main_sub)
+            else
+                local temp_job = find_temp_job(self)            
+                windower.add_to_chat(4,"JobChange: Conflict with "..conflict)
+                if main_sub == conflict then 
+                    jobchange(temp_job,main_sub)
+                    jobchange(jobid,main_sub)
+                else
+                    jobchange(temp_job,conflict)
+                    jobchange(jobid,main_sub)
+                end
+            end
+        else
+            windower.add_to_chat(4,"JobChange: Not close enough to a Moogle!")
+        end        
+    else
+        windower.add_to_chat(4,"JobChange: Could not change "..command.." to "..job:upper().." ---Mistype|NotUnlocked")
+    end
+    
+end)
+
+function find_conflict(job,self)
+    if self.main_job == job:upper() then
+        return "main"
+    end
+    if self.sub_job == job:upper() then
+        return "sub"
+    end
+end
+
+function find_temp_job(self)
+    local starting_jobs = S { 'WAR', 'MNK', 'WHM', 'BLM', 'RDM', 'THF' } 
+    for i in pairs(starting_jobs) do
+        if not find_conflict(i,self) then 
+            for index,value in pairs(res.jobs) do
+                if value.ens == i then
+                    return index
+                end
+            end
```

Comment:
This loop can be replaced by a helper function from the resources library. You need to find a job with a certain short name. The resource library provides a mechanism to search for a specific resource with:
```lua
    for job in starting_jobs:it() do -- same as above
        if not find_conflict(job, self) then -- same as above
            return res.jobs:with('english_short', job).id -- inner loop replaced with this
        end
    end
```

Now, you could also exploit that the starting jobs have the IDs 1 through 6, and use another helper mechanism of the resources library:
```lua
funtion find_temp_job(self)
    -- This creates a list of starting jobs, including all the fields from the resources (id, name, english, english_short, etc.)
    local starting_jobs = res.jobs:id(function(id) return id >= 1 and id <= 6 end)
    -- This applies a filter function to the list of starting jobs. The result is a list of all starting jobs that don't have a conflict)
    local valid_jobs = starting_jobs:filter(function(job) return not find_conflict(job, self) end)
    -- Then you just select the first one of those and return its ID (which is safer to go for than the index)
    return valid_jobs[1].id
end
```